src/protocol_version.rs

```diff
@@ -0,0 +1,37 @@
+use std::sync::{Once, ONCE_INIT};
+use std::hash::Hasher;
+
+use crc::Hasher32;
+use crc::crc32;
+
+pub use net::constants::PROTOCOL_VERSION;
+
+static CALCULATE_CRC32: Once = ONCE_INIT;
+static mut VERSION_CRC32: u32 = 0;
+
+pub struct ProtocolVersion
+{
+    crc_32: Option<u32>
+}
+
+impl ProtocolVersion
+{
+    pub fn get_version() -> &'static str
+    {
+        return PROTOCOL_VERSION
+    }
+
+    /// This will return the crc32 from the current protocol version.
+    pub fn get_crc32() -> u32 {
+        CALCULATE_CRC32.call_once(|| {
+            unsafe { VERSION_CRC32 = crc32::checksum_ieee(PROTOCOL_VERSION.as_bytes()); }
```

Comment:
So as I have learned if we call unsafe code and wrap it, we need to have some invariants that we check that guarantee that this is actually safe.

Reply:
I am using `lazy_static` instead now. Think this is more cleaner.